Src/AutoNSubstituteUnitTest/SubstituteRelayTest.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using NSubstitute;
+using NSubstitute.Exceptions;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.TestTypeFoundation;
+using Xunit;
+using Xunit.Extensions;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class SubstituteRelayTest
+    {
+        [Fact]
+        public void ClassImplementsISpecimenBuilderToServeAsResidueCollector()
+        {
+            // Fixture setup
+            // Exercise system
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteRelay)));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateThrowsArgumentNullExceptionWhenContextIsNullBecauseItsRequired()
+        {
+            // Fixture setup
+            var sut = new SubstituteRelay();
+            var request = new object();
+            // Exercise system
+            var e = Assert.Throws<ArgumentNullException>(() => sut.Create(request, null));
+            // Verify outcome
+            Assert.Equal("context", e.ParamName);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotAType()
+        {
+            // Fixture setup
+            var sut = new SubstituteRelay();
+            object request = "beer";
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object result = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(result);
+            Assert.Same(request, noSpecimen.Request);

```

Comment:
`NoSpecimen` has structural equality, so it'd be easier to read the assertion if it was expressed like this:

```C#
var expected = new NoSpecimen(request);
Assert.Equal(expected, specimen);
```